core/src/main/java/com/linecorp/armeria/client/WebClient.java

```diff
@@ -312,15 +352,50 @@ static WebClientBuilder builder(URI uri) {
         return new WebClientBuilder(uri);
     }
 
+    /**
+     * Returns a new {@link WebClientBuilder} created with the specified {@code protocol}
+     * and base {@link EndpointGroup}.
+     *
+     * @throws IllegalArgumentException if the {@code protocol} is not one of the fields
+     *                                  in {@link SessionProtocol}
+     */
+    static WebClientBuilder builder(String protocol, EndpointGroup endpointGroup) {
+        return builder(SessionProtocol.of(requireNonNull(protocol, "protocol")), endpointGroup);
+    }
+
     /**
      * Returns a new {@link WebClientBuilder} created with the specified {@link SessionProtocol}
      * and base {@link EndpointGroup}.
      *
+     * @throws IllegalArgumentException if the {@code protocol} is not one of the fields
+     *                                  in {@link SessionProtocol}
+     */
+    static WebClientBuilder builder(SessionProtocol protocol, EndpointGroup endpointGroup) {
+        return builder(protocol, endpointGroup, null);
+    }
+
+    /**
+     * Returns a new {@link WebClientBuilder} created with the specified {@code protocol}.
+     * base {@link EndpointGroup} and path.
+     *
+     * @throws IllegalArgumentException if the {@code protocol} is not one of the fields
+     *                                  in {@link SessionProtocol}
+     */
+    static WebClientBuilder builder(String protocol, EndpointGroup endpointGroup, @Nullable String path) {
+        return builder(SessionProtocol.of(requireNonNull(protocol, "protocol")),
+                       endpointGroup, path);
+    }
+
+    /**
+     * Returns a new {@link WebClientBuilder} created with the specified {@link SessionProtocol},
+     * base {@link EndpointGroup} and path.
+     *
      * @throws IllegalArgumentException if the {@code sessionProtocol} is not one of the fields
      *                                  in {@link SessionProtocol}
      */
-    static WebClientBuilder builder(SessionProtocol sessionProtocol, EndpointGroup endpointGroup) {
-        return new WebClientBuilder(sessionProtocol, endpointGroup);
+    static WebClientBuilder builder(SessionProtocol sessionProtocol, EndpointGroup endpointGroup,
```

Comment:
nit: Change `sessionProtocol` to `protocol` for consistency?
```suggestion
    static WebClientBuilder builder(SessionProtocol protocol, EndpointGroup endpointGroup,
```